pkg/deploy/util/util.go

```diff
@@ -390,6 +397,35 @@ func int32AnnotationFor(obj runtime.Object, key string) (int32, bool) {
 	return int32(i), true
 }
 
+// DeploymentsForCleanup determines which deployments for a configuration are relevant for the
+// revision history limit quota
+func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployments []api.ReplicationController) []api.ReplicationController {
```

Comment:
I don't see any utility in this function since I cannot imagine where else is it useful except for the controller but anyway if we are going to have it here I would expect it to return the deployments I need to delete ie. it would also include the zero check.